### PR TITLE
Added the up to date feature/spacerip utils into master so we can rebase

### PR DESCRIPTION
**Pull request by Sorrento110**

### CODE REVIEW

File: api/src/utils.py [hunk 1]
```diff
@@ -35,46 +51,110 @@ def delete_matching_records_from_model(model_id, record_key, record_test):
...
+        "run_id": run.id,
+    }
     es.index(index="tests", body=body, id=model_id)
 
-    return run_id    
\ No newline at end of file
+    return run_id
+
+
+def get_rawfile(uuid, filename):
```
Comment (fivegrant) [inline api/src/utils.py]: Does get and put in scenario take a name with the full path?

**Comment by Sorrento110** [comment]
put_rawfile will now accept a file prefix argument that allows it to create parent directories before the uuid directory on both local development storage and S3.

**Comment by fivegrant** [comment]
file_prefix is in the put but not the get. Does that mean the get expects the filename with the full path, while they are separated on the put side? Would it be all right to have a similar interface for both. I don't think any changes to get_rawfile would be needed if put_rawfile extracted the path from the filename.

**Comment by Sorrento110** [comment]
@fivegrant that's valid, let me add it.

**Comment by fivegrant** [comment]
```
# Could we add something like this into `put_rawfile`?

if filename.startswith("/"):
  filename = filename[1:] 
filehandle, file_prefix = (os.path.basename(filename), os.path.dirname(filename)

# Replace all instances of `filename` after this to `filehandle`
```

File: api/src/utils.py (hunk 1 above)
Comment (fivegrant) [reply api/src/utils.py]: This has been changed. Whole path can now be given

**Comment by fivegrant** [comment]
I think this is pretty much ready to merge. Model registration and running are still functional with shortrip. Made a commit that fixed an unreferenced varaible, but you might want to double check the scenario of creating a None file @Sorrento110

**Comment by fivegrant** [comment]
Just realized that `api/src/settings` is a dependency. I don't think this will work without it `DATASET_STORAGE_BASE_URL` existing in there

**Comment by fivegrant** [comment]
LGTM?

**Review by fivegrant — APPROVED** [review]

**Pull request merged**